client/utils/utils.go

```diff
@@ -0,0 +1,56 @@
+package utils
+
+import (
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/client/keys"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authctx "github.com/cosmos/cosmos-sdk/x/auth/client/context"
+)
+
+// SendTx implements a auxiliary handler that facilitates sending a series of
+// messages in a signed transaction given a TxContext and a QueryContext. It
+// ensures that the account exists, has a proper number and sequence set. In
+// addition, it builds and signs a transaction with the supplied messages.
+// Finally, it broadcasts the signed transaction to a node.
+func SendTx(txCtx authctx.TxContext, queryCtx context.QueryContext, msgs []sdk.Msg) error {
+	if err := queryCtx.EnsureAccountExists(); err != nil {
+		return err
+	}
+
+	from, err := queryCtx.GetFromAddress()
+	if err != nil {
+		return err
+	}
+
+	if txCtx.AccountNumber == 0 {
```

Comment:
This really isn't the right check - what if the account number was intentionally tried as `0`? I think we want to check whether or not the user passed the `--account-number` flag at all.

Reply:
Hmmm, it seems to be identical to the logic that currently exists:

https://github.com/cosmos/cosmos-sdk/blob/develop/client/context/viper.go#L105-L122

If the client does indeed intentionally pass `0`, then it'll look it up automatically -- kind of quirky, I understand.

If, as you're suggesting, we want to check if the client passed `0` and actually go with that, then'll be need to introduce the dependency of viper here. Thoughts?

Reply:
Can't we just change the default value to like `-1` to mean automatic?

Reply:
Yeah, but then the UX will be a little weird cause the CLI description will have the default in there of `-1`.

Let me think on this a bit more -- I can probably come up with a cleaner way outside the scope of `utils`.

Reply:
Sounds good! I'd actually recommend just putting a "TODO: <problem>" above the relevant lines / moving this to a new issue (since this isn't a performance regression). We can do these UI `nice to haves` until later! (I doubt practically anyone will be doing --sequence=0 for the next month)

Reply:
Actually, fair point! I'll do this :-)